### ISSUE TITLE
CCv0: Improve the bundle merge process between pod config and container image config

### ISSUE THREAD

**Issue by arronwy** [kata-containers/kata-containers#4323]
Since container image is pulled inside the guest, we need merge the image bundle OCI spec with the container creation request OCI spec which implemented in commit: ab6b1cbfe94667b54d14f4f72fbd9fa7b1538b2a
But as describe in Kubernetes spec:
https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#entrypoint
for entrypoint container image's ENTRYPOINT is used if this is not provided. 
for workingDir  If not specified, the container runtime's default will be used,

Current impelemtation checks whether container creation request is empty, not sure when these options is not configured in the pod yaml file but contained will give a default value which will prevent we load the config from container image. 
```
if target.args.is_empty() && !source.args.is_empty() {
if target.cwd.is_empty() && !source.cwd.is_empty() {
target.env.append(&mut source.env.clone());
```
For the environment variable, we found only the Kubernetes related variable are added and defined in container image is not take effect by the kata agent.

**Comment by arronwy** [comment]
Hi @wllenyj could you have a look of this issue?

**Comment by wllenyj** [comment]
@arronwy Sorry for the late reply.
This is how containerd handles Command of OCI specs:

https://github.com/containerd/containerd/blob/main/pkg/cri/opts/spec.go#L54-L76

And handle CWD: 

https://github.com/containerd/containerd/blob/01a2b793b88459004d319b641b2fc751ab13f21c/pkg/cri/server/container_create_linux.go#L145-L149

We need to add Entrypoint compatibility.

**Comment by wllenyj** [comment]
A related question: https://github.com/containerd/containerd/issues/7229

**Comment by arronwy** [comment]
similar issue: https://github.com/kata-containers/kata-containers/issues/4828

**Comment by huoqifeng** [comment]
@wllenyj is this issue in progress already? I just found the issue after I created https://github.com/kata-containers/kata-containers/issues/4828 when try istio, it's obviously same problem.

**Comment by jeromedoucet** [comment]
If anyone agree, I will have a look at that :) .

**Comment by wllenyj** [comment]
@huoqifeng Yes, this is in progress. Do you have an image available there to verify this?

**Comment by huoqifeng** [comment]
@wllenyj here is an example: https://github.com/kata-containers/kata-containers/issues/4828#issuecomment-1214489203

**Comment by huoqifeng** [comment]
Priority of the command, entrypoint and args is described in https://stackoverflow.com/questions/59248318/kubectl-run-command-vs-arguments, and https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#notes

**Comment by wllenyj** [comment]
In Progress
https://github.com/wllenyj/containerd/commit/e8ddc536812452dbac265338ecd3378a8a8b1429
